cranelift/codegen/src/machinst/isle.rs

```diff
@@ -299,12 +299,12 @@ macro_rules! isle_prelude_methods {
 /// internally has a temporary reference to a machinst `LowerCtx`.
 pub(crate) struct IsleContext<'a, C: LowerCtx, F, I, const N: usize>
 where
-    [C::I; N]: smallvec::Array,
+    [(C::I, bool); N]: smallvec::Array,
 {
     pub lower_ctx: &'a mut C,
     pub flags: &'a F,
     pub isa_flags: &'a I,
-    pub emitted_insts: SmallVec<[C::I; N]>,
+    pub emitted_insts: SmallVec<[(C::I, bool); N]>,
```

Comment:
This probably increases memory usage by a considerable amount due to padding. Maybe we could instead have an `is_safepoint` method on instructions?

Reply:
This vector tends to be small, it only holds the machine instructions emitted for a single CLIF instruction.   So I don't think memory usage is really any concern here ...

Reply:
Right, forgot about that.